price_parser/parser.py

```diff
@@ -36,15 +37,36 @@ def fromstring(cls, price: Optional[str],
         which may contain currency, as a hint. If currency is present in
         ``price`` string, it could be **preferred** over a value extracted
         from ``currency_hint`` string.
+
+        ``thousand_with_dot`` is optional; it is used to determine if
+        thousand separator is a dot or a comma. If ``thousand_with_dot``
+        is ``True``, then ``1.000`` is parsed as ``1000``. If it is ``False``,
+        then ``1.000`` is parsed as ``1``.
         """
+        currency = extract_currency_symbol(price, currency_hint)
+        if currency is not None:
+            currency = currency.strip()
+        for currency_key in CURRENCIES.keys():
+            if CURRENCIES[currency_key]["s"] == currency:
+                if CURRENCIES[currency_key]["d"] == 3:
+                    thousand_with_dot = False
+                break
+            if CURRENCIES[currency_key]["sn"] == currency:
+                if CURRENCIES[currency_key]["d"] == 3:
+                    thousand_with_dot = False
+                break
+            alternative_symbols = CURRENCIES[currency_key].get("sn2", [])
+            if currency in alternative_symbols:
+                if CURRENCIES[currency_key]["d"] == 3:
+                    thousand_with_dot = False
+                break
         amount_text = extract_price_text(price) if price is not None else None
+        if not thousand_with_dot:
+            decimal_separator = '.'
```

Comment:
This seems wrong, as it overrides even if the user specified a given decimal separator.

I do like the idea of choosing a digit-group/decimal separator based on the other separator if one of them is not specified.